packages/lodestar/src/api/rest/routes/validator/publishBlock.ts

```diff
@@ -31,11 +31,13 @@ export const registerBlockPublishEndpoint = (fastify: IFastifyServer, modules: I
     "/block",
     opts,
     async (request, reply) => {
-      await modules.chain.receiveBlock(
-        modules.config.types.SignedBeaconBlock.fromJson(
-          request.body.beacon_block
-        )
+      const block = modules.config.types.SignedBeaconBlock.fromJson(
+        request.body.beacon_block
       );
+      await Promise.all([
+        modules.chain.receiveBlock(block),
+        modules.network.gossip.publishBlock(block)
```

Comment:
I'm thinking we should wait for the chain to process the block before publishing via gossip? (And let the "sync" module relay the block to the network)

Reply:
But we will then publish blocks that we are syncing and potentially ruin our peer reputation

Reply:
I guess my comment is more about wondering if there is a way to hide network internals from the API, or at least this part of the code.
IMO the logic here should look something like:
"push new block into the system"
rather than
"add block here, add block there, etc"

Reply:
Probably, but api is only entrypoint where block needs to be processed as well as gossiped :/

Reply:
Ok, how about this goes to `src/api/impl`?

Reply:
yes, but I'm gonna do it in separate PR where I will drop rpc?